test/onigumo_test.exs

```diff
@@ -20,18 +24,65 @@ defmodule OnigumoTest do
       end
     )
 
-    assert(:ok == Onigumo.download(HTTPoisonMock, @url))
-    assert("Body from: #{@url}" == File.read!(@filename))
+    url = Enum.at(@urls, 0)
+    assert(:ok == Onigumo.download(HTTPoisonMock, url))
+    assert("Body from: #{url}" == File.read!(@filename))
   end
 
 
+  test("download multiple URLs") do
+    expect(
+      HTTPoisonMock,
+      :get!,
+      length(@urls),
+      fn url ->
+        %HTTPoison.Response{
+          status_code: 200,
+          body: "Body from: #{url}"
+        }
+      end
+    )
+
+    responses = Enum.map(@urls, fn _ -> :ok end)
+    assert(responses == Onigumo.download(HTTPoisonMock, @urls))
+
+    last_url = Enum.at(@urls, -1)
+    assert("Body from: #{last_url}" == File.read!(@filename))
+  end
+
+  test("download URLs from the input file") do
+    expect(
+      HTTPoisonMock,
+      :get!,
+      length(@urls),
+      fn url ->
+        %HTTPoison.Response{
+          status_code: 200,
+          body: "Body from: #{url}"
+        }
+      end
+    )
+
+    content = Enum.map(@urls, &(&1 <> " \n")) |> Enum.join()
```

Comment:
That's little bit confusing at first sight. But the benefit is obvious against the readability,  I supposed...

Reply:
```elixir
fn url -> url <> " \n" end
```
or

```elixir
fn url -> "#{url} \n" end
```

may be more readable. I agree that it’s not the nicest part here. ’d however say that the `&()` lambdas are ok if you are familiar with them. But this may not remain here as with more tests it is a hot candidate to be DRIED. See #44.